src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp

```diff
@@ -66,27 +505,397 @@ class boss_thorim : public CreatureScript
+                if (Creature* pillar = ObjectAccessor::GetCreature(*me, _activePillarGUID))
+                    pillar->RemoveAllAuras();
+
+                if (_hardMode)
+                {
+                    if (Creature* sif = instance->GetCreature(DATA_SIF))
+                    {
+                        summons.Despawn(sif);
+                        sif->DespawnOrUnsummon(10000);
+                    }
+                }
+
                 _JustDied();
+
+                Talk(SAY_DEATH);
+                events.ScheduleEvent(EVENT_OUTRO_1, 4000);
+                events.ScheduleEvent(EVENT_OUTRO_2, _hardMode ? 8000 : 11000);
+                events.ScheduleEvent(EVENT_OUTRO_3, _hardMode ? 19000 : 21000);
+
+                me->m_Events.AddEvent(new KeeperDespawnEvent(me), me->m_Events.CalculateTime(35000));
             }
 
             void EnterCombat(Unit* /*who*/) override
             {
-                Talk(SAY_AGGRO);
                 _EnterCombat();
+                Talk(SAY_AGGRO_1);
+
+                events.SetPhase(PHASE_1);
+
+                events.ScheduleEvent(EVENT_SAY_AGGRO_2, 9000, 0, PHASE_1);
+                events.ScheduleEvent(EVENT_SAY_SIF_START, 16500, 0, PHASE_1);
+                events.ScheduleEvent(EVENT_START_SIF_CHANNEL, 22500, 0, PHASE_1);
+
+                events.ScheduleEvent(EVENT_STORMHAMMER, 40000, 0, PHASE_1);
+                events.ScheduleEvent(EVENT_CHARGE_ORB, 30000, 0, PHASE_1);
+                events.ScheduleEvent(EVENT_SUMMON_ADDS, 15000, 0, PHASE_1);
+                events.ScheduleEvent(EVENT_BERSERK, 369000);
+
+                DoCast(me, SPELL_SHEATH_OF_LIGHTNING);
+
+                if (Creature* runicColossus = instance->GetCreature(DATA_RUNIC_COLOSSUS))
+                {
+                    runicColossus->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                    runicColossus->AI()->DoAction(ACTION_ACTIVATE_ADDS);
+                }
+
+                if (GameObject* lever = instance->GetGameObject(DATA_THORIM_LEVER))
+                    lever->RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_NOT_SELECTABLE);
+
+                // Summon Sif
+                me->SummonCreature(NPC_SIF, SifSpawnPosition);
+            }
+
+            void JustSummoned(Creature* summon) override
+            {
+                switch (summon->GetEntry())
+                {
+                    case NPC_LIGHTNING_ORB:
+                    {
+                        summon->SetReactState(REACT_PASSIVE);
+                        summon->CastSpell(summon, SPELL_LIGHTNING_DESTRUCTION, true);
+
+                        summon->GetMotionMaster()->MovePoint(EVENT_CHARGE_PREPATH, LightningOrbPath[LightningOrbPathSize - 1], false);
+
+                        Movement::PointsArray path(LightningOrbPath, LightningOrbPath + LightningOrbPathSize);
+
+                        Movement::MoveSplineInit init(summon);
+                        init.MovebyPath(path);
+                        init.Launch();
```

Comment:
Don't have sniff for electric ball of DOOM (tm), the waypoints are joschiwald's work